gpustat.py

```diff
@@ -262,22 +263,22 @@ def running_processes():
                                   })
             process_entries.append(process_entry)
 
-        pid_map = {int(e['pid']) : None for e in process_entries
-                   if not 'Not Supported' in e['pid']}
-
-        # 2. map pid to username, etc.
-        if pid_map:
-            pid_output = execute_process('ps -o {} -p {}'.format(
-                'pid,user:16,comm',
-                ','.join(map(str, pid_map.keys()))
-            ))
-            for line in pid_output.split('\n'):
-                if (not line) or 'PID' in line: continue
-                pid, user, comm = line.split()[:3]
-                pid_map[int(pid)] = {
-                    'user' : user,
-                    'comm' : comm
-                }
+        pid_map = {int(e['pid']) : None for e in process_entries if not 'Not Supported' in e['pid']}
+        if os.environ.get('DOCKER_CONTAINER') is  None:
```

Comment:
The existence of docker should be transparent to the code itself, IMHO.

Reply:
raceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1982, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1614, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1517, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1612, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python2.7/dist-packages/flask/app.py", line 1598, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "flask_app.py", line 18, in get_status
    data = gpustat.print_gpustat(**{'show_cmd': False, 'gpuname_width': 16, 'show_user': False, 'show_pid': False, 'json': True, 'no_color': False, 'ret_json': True})
  File "/home/stats/gpustat.py", line 414, in print_gpustat
    gpu_stats = GPUStatCollection.new_query()
  File "/home/stats/gpustat.py", line 246, in new_query
    return GPUStatCollection(gpu_list)
  File "/home/stats/gpustat.py", line 219, in __init__
    self.update_process_information()
  File "/home/stats/gpustat.py", line 310, in update_process_information
    processes = self.running_processes()
  File "/home/stats/gpustat.py", line 274, in running_processes
    ','.join(map(str, pid_map.keys()))
  File "/home/stats/gpustat.py", line 47, in execute_process
    stdout = check_output(command_shell, shell=True).strip()
  File "/usr/lib/python2.7/subprocess.py", line 573, in check_output
    raise CalledProcessError(retcode, cmd, output=output)
CalledProcessError: Command 'ps -o pid,user:16,comm -p 110117,101614' returned non-zero exit status 1

the pids exist on the host, not in the container so nonzero exit script dies.